packages/core/src/shared/ui/common/skipPrompter.ts

```diff
@@ -3,24 +3,17 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { StepEstimator } from '../../wizards/wizard'
+import { StepEstimator, WIZARD_SKIP } from '../../wizards/wizard'
 import { Prompter, PromptResult } from '../prompter'
 
-/** Pseudo-prompter that immediately returns a value (and thus "skips" a step). */
+/** Prompter that return SKIP control signal to parent wizard */
 export class SkipPrompter<T> extends Prompter<T> {
-    /**
-     * @param val Value immediately returned by the prompter.
-     */
-    constructor(public readonly val: T) {
```

Comment:
The previous implementation supports resolving the value it was assigned in the beginning. but this is removed in the new solution. Is this intended?

Reply:
A straight forward case would be in `bindPrompter`, we have multiple if/else based on previous states, and maybe there are two case leads to skip prompter, but we want different value for this field even though they are skipped.

Reply:
The previous workaround for skipping prompter was to automatically resolve the value and skip displaying UI to customer. However, that did not work for backward direction since the auto resolve always revert the backward flow to forward flow.

So we are splitting the skipping case from setting default value case. For how to use the SkipPrompter, please follow this example:  [aws-toolkit-vscode/packages/core/src/dev/activation.ts at master · vicheey/aws-toolkit-vscode](https://github.com/vicheey/aws-toolkit-vscode/blob/master/packages/core/src/dev/activation.ts#L414)

Reply:
You can consider resolving the value when forward and skip when backwards right?